tests/unit/acquisition/function/test_function.py

```diff
@@ -15,7 +15,7 @@
 from __future__ import annotations
```

Comment:
Do we also want to also add an integration test case to `test_bayesian_optimization.py`? (something similar to the notebook setup perhaps)

Reply:
This will take a bit longer to implement. I'll look into it and maybe add it as part of the next PR where we will have more optimisation methods.